### DISCUSSION TITLE
Artifacts aren't shown when a job fails

### DISCUSSION THREAD

**Discussion by jay** [community/community#108118] [Actions] [answered]
### Select Topic Area

Bug

### Body

curl-for-win makes daily builds and if all the jobs are [successful](https://github.com/curl/curl-for-win/actions/runs/7864514181) then a list of artifacts is shown:

![example1](https://github.com/community/community/assets/965580/9cd9c6ce-7d2b-4589-8b8d-857b38841d16)

If even a single job [fails](https://github.com/curl/curl-for-win/actions/runs/7877559459) then no artifacts even from successful jobs are shown:

![example2](https://github.com/community/community/assets/965580/a1f9447e-1970-4645-9f6e-a7a39037e5b5)

/cc @vszakats

**Comment by vszakats** [comment, marked as answer]
The reason for this is not the failed job, but that I did click the UI to retry that failed job, which creates a copy of the job results, which copy misses the artifacts already built.

To find those artifacts, you need to click on the `Latest attempt #2` button and select `Attempt #1`.
<img width="313" alt="Screen Shot 2024-02-13" src="https://github.com/community/community/assets/1446897/813fcc70-b44d-46ec-b0b6-bff42e63226e">

This doesn't look like something we can fix locally, besides refraining from using the "retry failed jobs" feature. In this case the failure turned out to be a package mixup in Debian sid RISC-V so retrying didn't help.

**Reply by jay** [reply]
That makes sense. Thanks.

**Reply by joshuapinter** [reply]
This is, in fact, the reason but I think the Artifacts should show all of the Artifacts uploaded for the entire job, not just a single Attempt. Multiple Attempts would overwrite things as necessary but if you're just retrying a single job all of a sudden your past artifacts disappear and you have to go to an earlier attempt to see them.
